packages/repack/src/webpack/loaders/assetsLoader.ts

```diff
@@ -293,3 +316,61 @@ export default async function reactNativeAssetsLoader(this: LoaderContext) {
     callback?.(error as Error);
   }
 }
+
+const decodeDataUriContent = (encoding: string, content: string) => {
+  const isBase64 = encoding === 'base64';
+  return isBase64
+    ? Buffer.from(content, 'base64')
+    : Buffer.from(decodeURIComponent(content), 'ascii');
+};
+
+function inlineAssetLoader(content: string | Buffer, resource: string) {
+  const ext = path.extname(resource);
+  const match = URIRegEx.exec(resource);
+  let resultMimeType: string | boolean;
+  let mimeType: string;
+  let parameters: string;
+  let encodedContent: string;
+  let encoding: boolean | string;
+  if (match) {
+    mimeType = match[1] || '';
+    parameters = match[2] || '';
+    encoding = match[3] || false;
+    encodedContent = match[4] || '';
+  }
+
+  if (mimeType! !== undefined) {
+    resultMimeType = mimeType + parameters!;
+  } else if (ext) {
+    resultMimeType = mimeTypes.lookup(ext);
+  }
+
+  if (typeof resultMimeType! !== 'string') {
+    throw new Error(
+      "DataUrl can't be generated automatically, " +
```

Comment:
This error is incorrect. What `generator.mimetype` are we talking about? `"asset/resource`?